### PR TITLE
R4 Final Review

### PR DESCRIPTION
**Pull request by amsmith-pro**
No major changes to interface/ICD.

Major cleanups, improvements to unit testing, adds an additional intersection check when the client confirms the itinerary.

### CODE REVIEW

**Comment by owlot** [comment]
## Documents
- [ ] ICD
  - [ ] Current interfaces (grpc & rest)
  - [ ] Current relevant document links
- [ ] SDD
  - [ ] Description of grpc & rest handler implementations
  - [ ] Current sequence diagrams where applicable
  - [ ] Current relevant document links
- [ ] CONOPS
  - [ ] Current information
  - [ ] Bibliography (if any citations)
- [ ] README (see svc-cargo for example)
  - [ ] Badges
  - [ ] Accurate installation instructions
  - [ ] Current links to other documents
- [ ] Any instance of :construction: should genuinely be under construction

## Code (not applicable to examples/ folder)
- [ ] No unhandled unwrap()s
- [ ] Accurate (!) code comments on all functions
- [ ] No `# Arguments` and `# Returns` in function comments
- [ ] Function arguments checked where possible
- [ ] Sufficient logger (info, warn, and debug) messages
- [ ] Sensible Types (unsigned vs. signed)
- [ ] No `#[allow(dead_code)]` without good reason
- [ ] End all error, info, and debug messages with punctuation (periods, etc.)
- [ ] No FIXMEs or TODOs without a projected release (`TODO R4: etc.`)

## REST API
- [ ] Return codes accurately documented with openapi
- [ ] Return codes make sense
- [ ] Sufficiently descriptive error messages
- [ ] Do NOT reflect a user's bad input back at them in replies.
    - i.e. "Warning: '{}' is not a valid entry." should instead be "Warning: invalid entry."
    - Use debug messages to see the invalid argument.

## Cargo.toml(s)
- [ ] Package versions correct
- [ ] No minor versions specified
- [ ] Links to crates.io package, not git branch
- [ ] Must have `license-file = "LICENSE"` under `[package]`

File: server/src/grpc/macros.rs [hunk 1]
```diff
@@ -1,4 +1,4 @@
 //! log macro's for gRPC logging
 
 use lib_common::log_macros;
-log_macros!("grpc");
+log_macros!("grpc", "app::grpc");
```
Comment (owlot) [inline server/src/grpc/macros.rs:4]: Not sure why the prefix is specifically provided here. The macro defaults to `app` as prefix if nothing is provided.
Providing `app::grpc` as the prefix as is done here results into log messages in the namespace `app::grpc::grpc` (the first param will be appended to the namespace as well)

File: server/src/lib.rs
Comment (owlot) [inline server/src/lib.rs:1]: server shutdown test can be added here too:
```
#[cfg(test)]
mod tests {
    use super::*;

    #[tokio::test]
    async fn test_server_shutdown() {
        ut_info!("start");

        let (shutdown_tx, shutdown_rx) = tokio::sync::oneshot::channel::<()>();
        let (_, health_service) = tonic_health::server::health_reporter();
        tokio::spawn(async move {
            let _ = tonic::transport::Server::builder()
                .add_service(health_service)
                .serve_with_shutdown(
                    "0.0.0.0:50051".parse().unwrap(),
                    shutdown_signal("grpc", Some(shutdown_rx)),
                )
                .await;
        });

        // Send server the shutdown request
        assert!(shutdown_tx.send(()).is_ok());

        ut_info!("success");
    }
}

```

File: server/src/grpc/macros.rs (hunk 1 above)
Comment (amsmith-pro) [reply server/src/grpc/macros.rs]: https://github.com/aetheric-oss/svc-scheduler/pull/53/commits/8828aa5409cc759b31cd775b82e39728e8e4aba7

File: docs/sdd.md [hunk 2]
```diff
@@ -239,7 +239,7 @@ sequenceDiagram
     scheduler->>scheduler: main control loop<br>create_itinerary_impl(&task)
     scheduler->>+storage: search(...)<br>1 Aircraft, 2 Vertipads Information
     storage->>scheduler: Records for the aircraft and<br>vertipads in proposed itinerary
-    scheduler->>scheduler: Confirm that itinerary is possible
+    scheduler->>scheduler: Confirm that itinerary is possible;<br>check intersections with existing zones<br>and flight paths
```
Comment (owlot) [inline docs/sdd.md:242]: This syntax is giving an error in the GitHub view
Selection_425

Comment (amsmith-pro) [reply docs/sdd.md]: https://github.com/aetheric-oss/svc-scheduler/pull/53/commits/b9f523bb1322ac0661d75854c7e32764414a1b6a

Didn't like the semicolon

**Review by owlot — APPROVED** [review]

**Pull request merged**